lib/openstack_query/query_parser.py

```diff
@@ -73,36 +64,28 @@ def parse_group_by(
         :param include_missing: a flag which if set includes an extra grouping for values
         that don't fall into any group specified in group_ranges
         """
-        self._check_prop_valid(group_by)
+        if group_by not in self._prop_enum_cls:
+            raise ParseQueryError(
+                f"Error: Given property to group by: {group_by.name} is not supported by query"
+            )
         self._group_by = group_by
         all_prop_list = set()
 
         if group_ranges:
             logger.debug("creating filter functions for specified group ranges")
+            prop_func = self._prop_enum_cls.get_prop_func(group_by)
             for name, prop_list in group_ranges.items():
                 group_vals = tuple(prop_list)
-                self._group_mappings.update(
-                    {
-                        name: (
-                            lambda obj, lst=group_vals: self._prop_handler.get_prop(
-                                obj, group_by
-                            )
-                            in lst
-                        )
-                    }
+                self._group_mappings[name] = (
+                    lambda obj, lst=group_vals: prop_func(obj) in lst
```

Comment:
IMO each of this branches for `group_ranges should be extracted to a new method as there's enough there to warrant it:

```
if group_ranges:
    self._group_ranges(...)
```